### DISCUSSION TITLE
What Happened to the File Name Length Option

### DISCUSSION THREAD

**Discussion by MickeyMcNickname** [aclap-dev/video-downloadhelper#654] [Help and Q&A] [answered]
### Your question

I had set the file name length to 80 characters. But with the upgrade, I can't find that option and file name lengths seem to be reverted to 64 characters.

### Addon & CoApp version and details

version: 9.0.1.13
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"ed87fc2a57b34b9b"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "C:\\Users\\Downloads",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 5
}

**Comment by MTCMTC000** [comment]
I just found this problem too.  Maybe we could have the old settings pages back until the new version is complete.

**Reply by paulrouget** [reply, marked as answer]
I can add the settings back. I'm surprised people need such long names, but I'll do it.

**Reply by MTCMTC000** [reply]
The issue is not the desire to create the long file names but to have the entire original file name which likely contains significant information about the video.  Thank you

**Reply by paulrouget** [reply]
ok - I'll add it back :)

**Reply by GrampaWildWilly** [reply]
VDH picks up file names automatically from web pages.  Sometimes those names are long.  Personally, I manually shorten such ugly names when I can.  But like this user says, sometimes it just takes a lot of characters to contain all the meaningful identifying data.

**Reply by MTCMTC000** [reply]
The explanation below fixes my issue.  Thank you.

**Reply by chris0144** [reply]
> I can add the settings back. I'm surprised people need such long names, but I'll do it.
which version is the filename configurable in. 
I just LOVE the S/W, but prefered old GUI.
I am on version: 10.1.37.2
store: mozilla
lang: en-US
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:147.0) Gecko/20100101 Firefox/147.0
Internal storage: 480KB / 10240MB. 2 files
dev: false
jwt_store: ALL
valid_until: 1771943765
exp: 1777905365
entitlement_type: LIFETIME
user_id: 414423

**Reply by alexjalonso7777** [reply]
> I just found this problem too.  Maybe we could have the old settings pages back until the new version is complete.

Try it maybe might work who knows try it

**Reply by alexjalonso7777** [reply]
[]()https://github.com/alexjalonso7777

**Comment by GrampaWildWilly** [comment]
Or maybe we could just have certain settings added to the new settings dialog, which I find superior to the old one in every way.  There is the old setting that used to specify the width of the VDH menu in pixels.  I'd like that back.  I'd like a VDH setting that controlled the font & font size used in the VDH menu.  Those wouldn't require a reversion to the old settings approach.  Haven't you noticed that you no longer have to remember to click a Save button?  Don't you find it so much better that you click a setting & it's set, no further action required?  Some changes are improvements.  The settings in VDH 9 are an improvement.

**Reply by GrampaWildWilly** [reply]
Specifically for file names, I haven't looked into it deeply, being just another user, not an employee of VDH.  Is it just that the proposed file name on the VDH menu is truncated?  Once you get into the file selection dialog, what name is being proposed there?  Is it truncated as well?

**Comment by paulrouget** [comment]
Oh I'm stupid, we actually already support this:

## Go to smartnaming:

<img width="333" alt="Screenshot 2024-05-27 at 13 56 15" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/6e5f844c-d901-459a-a6d6-0df0dc3848a0">

## Change the name length for all websites:

<img width="693" alt="Screenshot 2024-05-27 at 13 56 31" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/3f39dc26-2c3f-4193-9a65-4ac9affcb5ce">

**Reply by jeff15110168** [reply]
how do i get the file be named based on the webpage title? This was working perfectly for me in the old version, now it's not working and i'm getting a bunch of files just named on the embedded mp4 site (i.e. 19050.mp4 instead of the useful webpage title).  %title is now not working...

**Reply by paulrouget** [reply]
Can you please share the URL of a page with that issue?

**Reply by jeff15110168** [reply]
I will PM you the site

**Reply by jeff15110168** [reply]
I guess there's no PM. files are being downloaded as 19090.mp4 instead of previously it was the title of site, even though %title is set in smartnaming. Is there a workaround or fix to make this work? Other sites seem to be pulling the title

**Reply by paulrouget** [reply]
Got it. I'll look into it (you can remove the gist if you want)

**Reply by paulrouget** [reply]
@jeff15110168 thank you for your help. I can reproduce the issues you mentioned. I'll try to fix it.

**Reply by jeff15110168** [reply]
thank you - having issues where the size is missing, the download bar is bouncing and the file is not pulling the title (3 issues that i commented on separate threads).

Appreciate you being able to help here :)

**Comment by MickeyMcNickname** [comment]
I am now using version Version 9.1.0.42. I have changed the smartnaming option to allow full length filenames. However, there is a new behavior present. I am now getting " - Youtube" added to each filename downloaded from Youtube. That was not how things worked prior to version: 9.0.1.13 where I simply changed the filename length.

For so many of your customers the added sitename like " - Youtube" is probably quite useful. But for me it isn't. So, is there any way to suppress the sitename and only get the video title?

**Reply by GrampaWildWilly** [reply]
My experience has been that YouTube names always contained ` - YouTube` at the end.  By always, I mean in VDH 8 & VDH 7 as well as now in VDH 9.  I haven't experimented with smartnaming, which @paulrouget Paul has fixed in the latest VDH beta.  Maybe there's a way to suppress the YouTube appendage on the file names.  If it's something I particularly want, I usually change the entire file name manually in the `Download As...` dialog anyway.  The proposed default is hardly ever exactly what I want.

**Reply by MickeyMcNickname** [reply]
I've only had 3 versions of this application, all since the free application someone else built broke. 
I don't know the first version, but the second was version: 9.0.1.13 and the current is version 9.1.0.42.

In the first version there was a setting to increase the file name length from a default of 64. And I did this. The result was a full length filename but not website tag. I was happy.

With 9.0.1.13 that option was removed; and I opened this thread in response. The app auto updated to 9.1.0.42, I checked back here and per instructions given to by paulrouget, I increased the file length using smart naming. The result was a full length file name and a website tag added to the filename. 

I am asking for instructions on how that website tag may be removed. Maybe it can't be in this version, so if that's so, that's also an answer.

**Reply by GrampaWildWilly** [reply]
You'll want to read this: #838.  There are examples in there of what to do.  They didn't work at that time but now they do.  I just tried a CSS %selector on a YouTube video.  XPath is no longer supported.  I highlighted only part of the video title but I don't think Firefox is quite that granular.  What I got when I did a Test in the smartnaming dialog was the full title, not just the part I highlighted.  But the tail of ` - YouTube` was omitted.  I then got into the file picker dialog & indeed, the name VDH proposed did not have ` - YouTube` on the end.  I'm not going to repeat all the steps here.  It's several screenshots, all of which I posted in that other thread.  You're going to have to get over tl;dr & just read the whole thing.

**Comment by MickeyMcNickname** [comment]
The solution in #838 does work on a site by site basis. However, I don’t see how truncating the filename is the default. That should be addressed at some future time.

Charles Hector
============
***@***.***

**Reply by GrampaWildWilly** [reply]
In #1132 I posted a bunch of screenshots showing how you can remove the " - YouTube" that gets appended to file names on YouTube.  You have to click one of those "Show x previous replies" things to see the explanation.  The appending of site identifiers like that is something each site does in its own weird way.  You would have to define a smartnaming rule for each site.  That's the way it works.  Every site codes their HTML in their own idiosyncratic way.  This isn't so much truncating the file name as it is choosing an alternative way of naming the file.  It may be tedious but it depends on how badly you want things to work.  You'll make the effort if you are sufficiently motivated.

Also, if you are going to interact with discussions here via E-mail, please remove the quoted part of what you are replying to.  Your posts are bloated up with duplicated posts.  That just clutters up the posts on the web site.  One copy of any given post is quite enough.

**Reply by MickeyMcNickname** [reply]
To: GrampaWildWilly
Thanks. I don't need any more help, so I won't be reading #1132. I used the CSS option you posted in #838 to get only the filename.

To: PaulRouget
In regard to truncating, I'm saying that without modifications to Version 9.1.0.42 it truncates the filename to 64 characters. I don't see that as optimal for most users. My guess is that most would like what the filename is on the website from which they are downloading. Those who don't can use "Download As" and modify. So I believe that the optimal solution is to make actual filename the default and add the option to truncate as a feature. Otherwise you are making users go through hoops to get what they  want.

Obviously as the add-on's creator, it's up to you to evaluate what your user base really wants. I've had my say, and I've implemented a fix. So for now I am happy.

**Comment by Trelligan** [comment]
### **Question:** Where can I find this 'smartnaming' area of the settings?
   I click the toolbar icon, click the bottom-left settings cogwheel, click the 'more settings' on the menu, and—no Smartnaming option. I'm sure it exists, I've seen screenshots, but I don't know how to get there. I want to change the filename length as on this thread.
   Do I need to switch to a Beta version?

Version 9.1.0.42 (stable), target mozilla, locale en-US. Firefox version 129.0.1 (64-bit). Windows 11, latest version.

**Reply by GrampaWildWilly** [reply]
![#01](https://github.com/user-attachments/assets/ce9b4984-b90f-4d90-a054-29ba8b1bc2e7)
![#02](https://github.com/user-attachments/assets/4d028664-1906-486c-8130-37a597558869)
![#03](https://github.com/user-attachments/assets/f8eefb51-c109-49b6-837f-85969369bdb7)
![#04](https://github.com/user-attachments/assets/6106c48c-9cc6-4a87-8cf8-4ef1b5a9ef02)
![#05](https://github.com/user-attachments/assets/40a1222f-4c0c-4b67-a53e-8275faf189fe)

I think it's a bit counterintuitive that you have to start by visiting a web page where VDH recognizes something.  You were making the egregious error of applying logic to the situation.  I've never broken myself of that terrible habit, either.  I have suggested to Paul that Smartnaming should not be captive to visiting a web page that has content.  Smartnaming _should_ be in the VDH settings somewhere.  Paul did say that was a good idea.  But this hasn't risen to the top of the priority list for him yet.  Oh well.  Until then, this is how you have to do it.

To @MickeyMcNickname I think what you are suggesting is that the default value for the maximum file name length should be larger than 64.  I say again, VDH is not actually truncating the name.  It's just applying this setting.  It's not intuitive how to change the setting.  That could be better.  But this is what you have to do now.  We can only hope things get more intuitive over time.

**Reply by MickeyMcNickname** [reply]
I don't see the distinction. When I wrote "without modifications to Version 9.1.0.42 it truncates the filename to 64 characters" That's exactly what happens for every filename greater than 64 characters. But if the nitpicking satisfies something within you, well, go ahead.

Thanks for the instructions you've posted. They were helpful.

**Comment by Trelligan** [comment]
### Bravo!
   Answer is exactly what I needed, delivered as concisely as possible.
